adapter/redisquota/redisquota_test.go

```diff
@@ -478,7 +481,9 @@ func TestBadConfig(t *testing.T) {
 
 func TestErrorResponse(t *testing.T) {
 	mockredis := testServer{errOnStart: false, errOnIntRet: true}
+	mockredis.Mutex.Lock()
```

Comment:
I'm surprised this works. What state is this actually protecting? Is this mockredis used anywhere else?

I would have expected some solution around:
- EITHER setting the port in the server address to ":0" (instead of "localhost:9736")
- OR having some global mutex that only allows one test at a time that sets up the `errOnIntRet` server.